include/cooking/cookingQuerys/ratingSystem.php

```diff
@@ -0,0 +1,136 @@
+            header('Location:/cookingWebsite/yourCookbook/addTags.php');
+}
+function deleteNewCookingTagQuery ($tagName){
+        dbQuery("DELETE FROM recipeTagIds WHERE recipeIngredientName=:recipeIngredientName", array('recipeIngredientName'=>$tagName));
+        dbQuery("DELETE FROM recipe_identifier_linked WHERE recipeIngredientName= :recipeIngredientName", array('recipeIngredientName'=>$tagName));
+        return header('Location:/cookingWebsite/yourCookbook/addTags.php');
+}
+
+// ********ALL FUNCTIONS FOR RATINGS BELOW*********
+
+function insertRatingIntoTable($loginId, $recipeId, $rating){
+    dbQuery("INSERT INTO ratings(loginId, recipeId, rating)
+            VALUES(:loginId, :recipeId, :rating)",
+            array('loginId'=>$loginId, 'recipeId'=>$recipeId, 'rating'=>$rating));
+}
+function algorithm(){
+    $total = 0;
+    $loginQuery = dbQuery("SELECT * FROM cookingLogin")->fetchALL();
+    $loginArray = array();
+        //$returnRecipeTags = "";
+        foreach ($loginQuery as $logins){
+            if(!@$loginArray[$logins['loginId']]){
+                $loginArray[$logins['loginId']] = true;
+            }
+                $loginId = $logins['loginId'];
+
+
+    $recipeQuery = dbQuery("SELECT * FROM recipes")->fetchALL();
+    $recipeArray = array();
+        //$returnRecipeTags = "";
+        foreach ($recipeQuery as $recipes){
+            if(!@$recipeArray[$recipes['recipeId']]){
+                $recipeArray[$recipes['recipeId']] = true;
+            }
+                $recipeId = $recipes['recipeId'];
+                $tagQuery = dbQuery("SELECT * FROM recipe_identifier_linked
+                            WHERE recipeId= :recipeId", array('recipeId'=>$recipeId))->fetchALL();
+                $tagArray = array();
+                foreach ($tagQuery as $tags){
+
+                    if(!@$tagArray[$tags['recipeIngredientName']]){
+                        $tagArray[$tags['recipeIngredientName']] = true;
+
+                    }
+                    $recipeIngredientName = $tags['recipeIngredientName'];
+                    $recipeTagQuery = dbQuery("SELECT * FROM recipe_identifier_linked
+                                        WHERE recipeId != :recipeId AND recipeIngredientName =:recipeIngredientName" ,
+                                        array('recipeId'=>$recipeId, 'recipeIngredientName'=>$recipeIngredientName))->fetchALL();
+                    $recipeTagArray = array();
+                    foreach ($recipeTagQuery as $recipeTag){
+                        if(!@$recipeTagArray[$recipeTag['recipeId']]){
+                            $recipeTagArray[$recipeTag['recipeId']] = true;
+                        }
+                        $recipeId2 = $recipeTag['recipeId'];
+                        $score=dbQuery("SELECT* FROM ratings WHERE recipeId = :recipeId AND loginId=:loginId", array('recipeId'=>$recipeId2, 'loginId'=>$loginId))->fetch();
+                        $total += $score['rating'];
```

Comment:
Does this work the way you expect? It looks like it might be possible to clean up the logic a bit. If it works, great, but it might be a good use of time to go in the conference room and walk through this algorithm from a high level to make sure it's doing what you want. I can also give some ideas on improving performance, although I understand you might not want to worry about that for now. But the performance improvements might also clean up the code.

Also, it looks like you're just totaling the scores which (I think) means you could end up with a predicted score > the max allowable score, and I think it means an ingredient with a bunch of 1-star ratings (i.e. you hate it) would perform better than an ingredient with a single five-star rating (i.e. you love it). But maybe I'm just misunderstanding how this works.